.github/workflows/release.yml

```diff
@@ -0,0 +1,63 @@
+---
+name: Release and Publish
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup python 🐍
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Install dependencies 👽
+        run: python -m pip install ansible-core yq rst-to-myst
+
+      - name: Verify release 🗒️
+        env:
+          TAG_NAME: ${{ github.ref_name }}
+        run: |
+          echo "::warning::TODO: confirm tests pass for this commit"
+          tag_version="${TAG_NAME##*v}"
+          version="$(yq -r .version galaxy.yml)"
+          if [[ ${tag_version} != ${version} ]]; then
+            echo "::error file=galaxy.yml::🔴 Tag version '${tag_version}' does not match galaxy.yml version '${version}'"
+            exit 1
+          fi
+          echo "::notice::🟢 Version number verified: '$version'"
+
+      - name: Build 🛠️
+        run: ansible-galaxy collection build -v --force .
+
+      - name: Prepare release notes 📓
+        env:
+          CHANGELOG_URL: "${{ format('{0}/{1}/blob/{2}/CHANGELOG.rst', github.server_url, github.repository, github.ref_name) }}"
+          REF_NAME: "${{ github.ref_name }}"
+        run: |
+          rst2myst stream --no-sphinx CHANGELOG.rst | csplit - "/^## ${REF_NAME}/" '/^## v.*/'
```

Comment:
I think here we will need to use also https://github.com/ansible-community/antsibull-changelog. Maybe whole CHANGELOG.rst will not be part of git repo, but only generated from ./changelogs/

Reply:
This part in the workflow assumes the changelog has already been updated using `antsibull-changelog`. I added that as a step to the release steps in the "[New release](https://github.com/ScaleComputing/HyperCoreAnsibleCollection/blob/main/.github/ISSUE_TEMPLATE/new-release.md)" issue template.

The process I threw together here is a bit disjointed. Here is how it should work right now:

1. Team decides there should be a release
2. Someone creates a "New release" issue for the version to be released and assigns it.
3. The assignee completes the release steps
    1. Update the version number in galaxy.yml e.g. `1.1.1`
    2. Update the changelog with `antsibull-changelog` e.g. `antsibull-changelog release --version 1.1.1`
    3. Tag the commit with the version number prefixed with 'v' e.g. `v1.1.1`
    4. Push the tag to github
4. Release workflow is triggered by the new tag.

I have a bunch of thoughts for improving this :laughing:

Reply:
OK, thanks for explaining, this sounds nice.